k8s-deployment/todo-app-service.yaml

```diff
@@ -5,6 +5,9 @@ metadata:
   namespace: todo-app
   labels:
     name: "todo-app"
+  annotations:
+    prometheus.io/scrape: "true"
+    prometheus.io/port: "3000"
```

Comment:
The prometheus target discovery via the kubernetes API allows both. Discovering pods or discovering all endpoints of a service as scrape target. I personally prefer discovering endpoints of services since you get the labels and annotations of the services attached to your prometheus metrics. 

Docs for kubernetes service discovery in prometheus: https://prometheus.io/docs/operating/configuration/#<kubernetes_sd_config>